amazon-kinesis-client/src/test/java/software/amazon/kinesis/coordinator/SchedulerTest.java

```diff
@@ -269,6 +274,24 @@ public final void testSchedulerShutdown() {
         verify(workerStateChangeListener, times(1)).onWorkerStateChange(WorkerStateChangeListener.WorkerState.SHUT_DOWN);
     }
 
+    @Test
+    public final void testErrorHandlerForUndeliverableAsyncTaskExceptions() {
+        AtomicBoolean wasHandlerInvoked = new AtomicBoolean(false);
+        Consumer<Throwable> testHandler = t -> wasHandlerInvoked.compareAndSet(false, true);
+
+        Scheduler schedulerSpy = spy(scheduler);
+
+        doAnswer(invocation -> {
```

Comment:
This is same as directly calling RxJavaPlugins.onError(). To address Yasemin's feedback, you need to mock the executor service to reject the task, when ops like subscribe() or onNext() is called. Similar to unit tests in my PR.

Reply:
Even when mocking the executor service to reject the tasks, it seems like the exceptions are occurring at the KCL layer still. In your unit tests, the exception from a mocked executor service is:
```java.util.concurrent.RejectedExecutionException```

Whereas using RxJavaPlugins.onError() gives what we're trying to trigger:
```io.reactivex.exceptions.UndeliverableException: java.util.concurrent.RejectedExecutionException: Test exception.```

I set up a failing executor service and got the same result as your tests, with the exception still happening in the KCL layer. I'm not sure it's possible to trigger an exception in the RxJava layer with mocks, or at least not easily. Curious to hear your thoughts?

Reply:
Leaving as is per in-person discussion on scope